src/librustc_mir/const_eval.rs

```diff
@@ -643,16 +643,21 @@ pub fn const_eval_raw_provider<'a, 'tcx>(
         let err = error_to_const_error(&ecx, error);
         // errors in statics are always emitted as fatal errors
         if tcx.is_static(def_id).is_some() {
-            let reported_err = err.report_as_error(ecx.tcx,
-                                                   "could not evaluate static initializer");
             // Ensure that if the above error was either `TooGeneric` or `Reported`
             // an error must be reported.
-            if tcx.sess.err_count() == 0 {
-                tcx.sess.delay_span_bug(err.span,
+            let reported_err = tcx.sess.track_errors(|| {
+                err.report_as_error(ecx.tcx,
+                                    "could not evaluate static initializer")
+            });
+            match reported_err {
+                Ok(v) => {
+                    tcx.sess.delay_span_bug(err.span,
                                         &format!("static eval failure did not emit an error: {:#?}",
-                                                 reported_err));
+                                        v));
+                    v
+                },
+                Err(_) => ErrorHandled::Reported,
```

Comment:
Please import https://doc.rust-lang.org/nightly/nightly-rustc/rustc/util/common/struct.ErrorReported.html in this module and use `ErrorReported` instead of `_`, so future changes to `track_errors` will get noticed here

Reply:
Hmm, commit 14b61f79d0c34899ed028b9dbcb5094d04078b26 couldn't be green because `ErrorReported` was unused. How do we use it?

Reply:
The problem was that you didn't have a `use` statement to import the type. Since it wasn't imported, the compiler thought you had a variable named `ErrorReported` instead of knowing that you meant the type. You can search in the rust source for existing `use` statements that mention `ErrorReported` and create one in this file.

Reply:
Oh, I got it! My fork is old so `ErrorReported` is already imported. But current Rust source doesn't have it. And I have mistake on Git, please wait...